### PR TITLE
[Doc] Clarify unsupported operations with DLS/FLS

### PR DESCRIPTION
**Pull request by ywangd**
When DLS/FLS is enabled on an index for a user, the user will not be
able to perform operations that can make data content accessible under a
different name. This is intentional. This PR makes it clear in the doc.

doc preview link: https://elasticsearch_89606.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/security-limitations.html#field-document-limitations

### CODE REVIEW

**Review by jakelandis — APPROVED** [review]
LGTM

File: x-pack/docs/en/security/limitations.asciidoc [hunk 1]
```diff
@@ -50,6 +50,12 @@ When a user's role enables document or <<field-level-security,field level securi
 * The user cannot perform write operations:
 ** The update API isn't supported.
 ** Update requests included in bulk requests aren't supported.
+* The user cannot perform operations that effectively make contents accessible under another name, including:
```
Comment (lockewritesdocs) [inline x-pack/docs/en/security/limitations.asciidoc:53]:
```suggestion
* The user cannot perform operations that effectively make contents accessible
under another name, including actions from the following APIs:
```

**Review by lockewritesdocs — APPROVED** [review]
Left a non-blocking suggestion if you want to use it, but otherwise LGTM 🦖

**Comment by ywangd** [comment]
@elasticmachine update branch

**Comment by ywangd** [comment]
@elasticmachine run elasticsearch-ci/part-2 run elasticsearch-ci/part-3

**Pull request merged**